Makefile.core.mk

```diff
@@ -262,10 +262,15 @@ fmt: format-go format-python tidy-go
 buildcache:
 	GOBUILDFLAGS=-i $(MAKE) -e -f Makefile.core.mk build
 
+# gobuild script uses custom linker flag to set the variables.
```

Comment:
https://github.com/istio/istio/blob/2a4a3889292bb9b8893ec69e352b8d4d2a5e329e/common/scripts/gobuild.sh#L46

I would just update the default ldflags here. stripping debug info should be the default.